src/server/task-list/index.njk

```diff
@@ -3,6 +3,60 @@
 
 {% extends 'layouts/page.njk' %}
 
+{% macro incompleteTasks() %}
+{% set incomplete = 0 %}
+  {% for row in tasks %}
+    {% if row.isValid %}
+      {% set incomplete = incomplete + 1 %}
+    {% endif %}
+  {% endfor %}
+
+  {{ incomplete }}
+{% endmacro %}
+
+{% macro createTaskList() %}
+
+  {% set summaryRows = [] %}
+  {% for row in tasks %}
+    {% if not row.isEnabled %}
+      {% set href = '' %}
+      {% set status = {
+        text: 'Cannot start yet',
+        classes: 'govuk-task-list__status--cannot-start-yet'
+      } %}
+    {% elif row.isValid %}
+      {% set href = row.summaryLink %}
+      {% set status = {
+        text: 'Completed',
+        classes: ''
+      } %}
+    {% else %}
+      {% set href = row.initialLink %}
+      {% set status = {
+        tag: {
+          text: 'Incomplete',
+          classes: 'govuk-tag--blue'
+        }
+      } %}
+    {% endif %}
+
+    {% set rows = (summaryRows, summaryRows.push({
+      title: {
+        text: row.title
+      },
+      href: href,
+      status: status
+    })) %}
+  {% endfor %}
+
+
+
+  {{ govukTaskList({
+    idPrefix: "task-list",
+    items: summaryRows
+  }) }}
+{% endmacro %}
```

Comment:
TBH, I think readability of this version has taken a hit - vs the readability of the javascript version it replaces.

Reply:
whilst is half see your point, i dont think the controller should know about css classes etc - that is entirely view logic and has no place anywhere but the view